packages/itwin/grouping-mapping-widget/src/test/GroupingMappingCustomUI.test.tsx

```diff
@@ -0,0 +1,185 @@
+    .returns(() => new SelectionChangeEvent());
+
+  selectionScopesManagerMock
+    .setup(async (x) => x.getSelectionScopes(connectionMock.object))
+    .returns(async () => []);
+  selectionManagerMock
+    .setup((x) => x.scopes)
+    .returns(() => selectionScopesManagerMock.object);
+
+  Presentation.setSelectionManager(selectionManagerMock.object);
+  await TestUtils.initializeUiFramework(connectionMock.object);
+  server.listen({
+    onUnhandledRequest: "warn",
+  });
+  server.printHandlers();
+});
+
+afterAll(() => {
+  server.close();
+});
+
+afterEach(() => {
+  server.resetHandlers();
+});
+
+describe("Groupings View with default UIs", () => {
+  it("List all groups and click on add group and edit group buttons", async () => {
+    // Arange
+    const mockGroups = groupsFactory();
+    mockMappingClient
+      .setup(async (x) => x.getGroups(moq.It.isAnyString(), moq.It.isAnyString(), moq.It.isAnyString()))
+      .returns(async () => mockGroups.groups);
+
+    // Act
+    const { user } = render(<Groupings mapping={mockMapping} goBack={jest.fn()} />);
+    await waitForElementToBeRemoved(() => screen.getByText(/loading/i));
+
+    // Assert
+    const addButton = screen.getAllByTestId("gmw-add-group-button");
+    expect(addButton).toHaveLength(1);
+
+    // Click on 'Add Group' button
+    await user.click(addButton[0]);
+
+    // Should have three menu items
+    const addSelection = screen.getAllByTestId("gmw-add-Selection");
+    expect(addSelection).toHaveLength(1);
+    const addSearch = screen.getAllByTestId("gmw-add-Search");
+    expect(addSearch).toHaveLength(1);
+    const addManual = screen.getAllByTestId("gmw-add-Manual");
+    expect(addManual).toHaveLength(1);
+
+    // Should have the right group number
+    const horizontalTiles = screen.getAllByTestId("gmw-horizontal-tile");
+    expect(horizontalTiles).toHaveLength(mockGroups.groups.length);
+
+    horizontalTiles.forEach((horizontalTile, index) => {
+      const groupTile = within(horizontalTile);
+      expect(
+        groupTile.getByText(
+          mockGroups.groups[index].groupName
+        )
+      ).toBeInTheDocument();
+      expect(
+        groupTile.getByText(
+          mockGroups.groups[index].description ?? ""
+        )
+      ).toBeInTheDocument();
+    });
+
+    // Click on first group more icon
+    const moreButton = screen.getAllByTestId("gmw-more-button");
+    expect(moreButton).toHaveLength(mockGroups.groups.length);
+
+    await user.click(moreButton[0]);
+
+    // Should have 3 context menu items
+    const contextMenuItems = screen.getAllByTestId("gmw-context-menu-item");
+    expect(contextMenuItems).toHaveLength(3);
+    expect(contextMenuItems[0]).toHaveTextContent("Edit");
+    expect(contextMenuItems[1]).toHaveTextContent("Properties");
+    expect(contextMenuItems[2]).toHaveTextContent("Remove");
+
+    // Hover on 'Edit'
+    await user.hover(contextMenuItems[0]);
+
+    // Should have 3 sub menu items
```

Comment:
Do I understand correctly that this is a temporary workaround until we have the group query creation metadata stored as a comment on the query itself?

Reply:
yes